src/memory/mod.rs

```diff
@@ -23,18 +37,42 @@ impl MemoryManagers {
     }
 }
 
-pub trait MemoryManager {
+impl<T: MemoryManagerUpdate> MemoryManager<T> {
+    fn ready_for_updates(&mut self, _ctx: &StateContext) -> bool {
+        if let Some(class) = self.manager.singleton {
+            if class.class == 0 {
+                return false;
+            }
+
+            return true;
+        }
+        false
```

Comment:
Oops. This line was actually different for the two managers. Not sure what the implications are of that @Eein . The code still runs.

Reply:
I feel like i fix that with every pr. And it still never gets updated. Should be the same